File: API/Entities/Key.cs
using System;
using API.Enums;

namespace API.Entities;

public class Key
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public required KeyPriceType PriceType { get; set; } = KeyPriceType.TYPE1;
    public required string SilcaCode { get; set; }
    public string ErrebiCode { get; set; } = "";
    public string JmaCode { get; set; } = "";
    public required int Price { get; set; }
    public required int Quantity { get; set; }
}

File: CLIENT/src/types/Key.ts
export interface Key {
    id: string;
    priceType: KeyPriceType;
    silcaCode: string;
    errebiCode: string;
    jmaCode: string;
    price: number;
    quantity: number;
    isArchived: boolean,
    archivedAt: Date,
    images: KeyImage[];
}

export interface KeyImage {
    id: number;
    url: string;
    publicId: string;
    keyId: string;
    isMain: boolean;
}

export enum KeyPriceType {
    TYPE1 = 1,
    TYPE2 = 2,
    TYPE3 = 3
}
